packages/dev/sharedUiComponents/src/lines/optionsLineComponent.tsx

```diff
@@ -80,44 +83,103 @@ export class OptionsLineComponent extends React.Component<IOptionsLineComponentP
+                        <input
+                            type="text"
+                            placeholder="Enter a custom font here"
+                            id="customFont"
+                            onKeyDown={(event) => {
+                                event.keyCode === 13 && this.props.addVal != undefined
+                                    ? (this.props.addVal(
+                                          { label: (document.getElementById("customFont") as HTMLInputElement).value, value: this.props.options.length + 1 },
+                                          Number(this.state.value)
+                                      ),
+                                      this.updateCustomValue())
+                                    : null;
+                            }}
+                            onBlur={() => {
+                                this.props.addVal != undefined
+                                    ? (this.props.addVal(
+                                          { label: (document.getElementById("customFont") as HTMLInputElement).value, value: this.props.options.length + 1 },
+                                          Number(this.state.value)
+                                      ),
+                                      this.updateCustomValue())
+                                    : null;
+                            }}
+                        />
+                    </div>
                 </div>
-                <div className="options">
-                    <select onChange={(evt) => this.updateValue(evt.target.value)} value={this.state.value ?? ""}>
-                        {this.props.options.map((option, i) => {
-                            return (
-                                <option selected={option.selected} key={option.label + i} value={option.value} title={option.label}>
-                                    {option.label}
-                                </option>
-                            );
-                        })}
-                    </select>
+            );
+        } else {
+            return (
+                <div className={"listLine" + (this.props.className ? " " + this.props.className : "")}>
+                    {this.props.icon && <img src={this.props.icon} title={this.props.iconLabel} alt={this.props.iconLabel} color="black" className="icon" />}
+                    <div className="label" title={this.props.label}>
+                        {this.props.label}
+                    </div>
+                    <div className="options">
+                        <select onChange={(evt) => this.updateValue(evt.target.value)} value={this.state.value === -1 || null || undefined ? 1 : this.state.value}>
+                            {this.props.options.map((option, i) => {
+                                return (
+                                    <option selected={option.selected} key={option.label + i} value={option.value} title={option.label}>

```

Comment:
Are the option.labels unique? If so, there is no need to add the index on the key, and it might even cause problems: https://robinpokorny.com/blog/index-as-a-key-is-an-anti-pattern/

Reply:
option.labels are unique, but this isn't code that I wrote and I wanted to keep it as close as possible to the orignal. Would you recommend removing the index and keeping it as key = {option.label}?

Reply:
Hmmm let's keep as it is then so we don't change too much at a time